### ISSUE TITLE
Remove the proc_macro::Group around metavariables which are always a single token

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust#55557]
I have the following proc macro to print out tokens:

```rust
extern crate proc_macro;
use proc_macro::TokenStream;

#[proc_macro]
pub fn print_tts(input: TokenStream) -> TokenStream {
    println!("{:#?}", input);
    TokenStream::new()
}
```

And suppose I call it like this:

```rust
#![feature(proc_macro_hygiene)]

extern crate print;

macro_rules! print_ident {
    ($i:ident) => {
        print::print_tts!(2 * $i)
    };
}

fn main() {
    print_ident!(ident);
}
```

As of rustc 1.31.0-nightly (1cf82fd9c 2018-10-30) the input is passed to my macro as:

```rust
TokenStream [
    Literal {
        lit: Integer(
            2
        ),
        suffix: None,
        span: #2 bytes(127..128)
    },
    Punct {
        ch: '*',
        spacing: Alone,
        span: #2 bytes(129..130)
    },
    Group {
        delimiter: None,
        stream: TokenStream [
            Ident {
                ident: "ident",
                span: #0 bytes(174..179)
            }
        ],
        span: #2 bytes(131..133)
    }
]
```

I believe idents (and any other metavariables which are always a single token) should not be wrapped in a Group. It makes handling proc macro input unnecessarily confusing. This proc macro should be receiving a plain `Literal` `Punct` `Ident` rather than `Literal` `Punct` `Group`.

Metavariables that can be multiple tokens, including $:expr and $:ty, will need to continue having a surrounding None-delimited Group to preserve precedence.

cc @alexcrichton

**Comment by petrochenkov** [comment]
FWIW, libsyntax does the same trick for identifiers and lifetimes (literals may contain minus, so they are not always single-token).
Lifetimes are not single-token in proc macros, so this leaves only idents.

**Comment by alexcrichton** [comment]
This makes sense to me as well, no need to place idents in a group!

**Comment by dtolnay** [comment]
Fixed by #92472.
